test_project/test_app/tests/test_storage.py

```diff
@@ -4,12 +4,80 @@
 from test_app import calc_info
-from autocompleter import Autocompleter, registry, signal_registry
+from autocompleter import base, Autocompleter, registry, signal_registry
 from autocompleter import settings as auto_settings
 
 
+class FacetedStoringAndRemovingTestCase(AutocompleterTestCase):
+    fixtures = ['stock_test_data_small.json']
+
+    def test_store_facet_data(self):
+        """
+        Storing saves facet data
+        """
+        aapl = Stock.objects.get(symbol='AAPL')
+        provider = FacetedStockAutocompleteProvider(aapl)
+        provider.store()
+
+        provider_name = provider.get_provider_name()
+        # the FacetedStockAutocompleteProvider get_facets is set to ['sector']
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Technology',)
+        keys = self.redis.keys(facet_set_name)
+        self.assertEqual(len(keys), 1)
+
+        facet_map_name = base.FACET_MAP_BASE_NAME % (provider_name,)
+        keys = self.redis.keys(facet_map_name)
+        self.assertEqual(len(keys), 1)
+
+        facet_data = provider._deserialize_data(self.redis.hget(keys[0], aapl.id))
+        self.assertEqual(facet_data, [{'key': 'sector', 'value': aapl.sector}])
+
+    def test_second_store_removes_old_facet_data(self):
+        """
+        Store removes outdated facet data
+        """
+        aapl = Stock.objects.get(symbol='AAPL')
+        provider = FacetedStockAutocompleteProvider(aapl)
+        provider.store()
+
+        provider_name = provider.get_provider_name()
+        # the FacetedStockAutocompleteProvider get_facets is set to ['sector']
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Technology',)
+        keys = self.redis.keys(facet_set_name)
+        self.assertEqual(len(keys), 1)
+
+        aapl.sector = 'Healthcare'
+        aapl.save()
+        provider.store()
+        # make sure the old key was removed after the second store call
+        keys = self.redis.keys(facet_set_name)
+        self.assertEqual(len(keys), 0)
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Healthcare',)
+        keys = self.redis.keys(facet_set_name)
+        self.assertEqual(len(keys), 1)
+
+    def test_remove_facet_data(self):
+        """
+        Remove takes care of deleting facet data
+        """
+        aapl = Stock.objects.get(symbol='AAPL')
+        provider = FacetedStockAutocompleteProvider(aapl)
+        provider.store()
+        provider.remove()
+
+        provider_name = provider.get_provider_name()
+        # the FacetedStockAutocompleteProvider get_facets is set to ['sector']
+        facet_set_name = base.FACET_SET_BASE_NAME % (provider_name, 'sector', 'Technology',)
+        keys = self.redis.keys(facet_set_name)
+        self.assertEqual(len(keys), 0)
+
+        facet_map_name = base.FACET_MAP_BASE_NAME % (provider_name,)
+        keys = self.redis.keys(facet_map_name)
+        self.assertEqual(len(keys), 0)
+
```

Comment:
I think it would behove us to so another test case that's test_store_all_facet_data() and test_remove_all_facet_data() that does the store_all() and see if it's the exact number of items you would expect in a one of the sector's facet sets and also the facet mapping over all